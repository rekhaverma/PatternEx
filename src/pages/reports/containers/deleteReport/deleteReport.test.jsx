import React from 'react';
import DeleteReport from './deleteReport.jsx';

describe('<DeleteReport /> modal: ', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      'onClose': jest.fn(),
      'deleteReport': jest.fn(),
      'data': {},
    };
    wrapper = shallow(<DeleteReport {...props} />);
  });

  describe('render: ', () => {
    it('should match with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('action simulation: ', () => {
    it('should call "deleteReport" function on click of delete', () => {
      wrapper.find('.button--success').simulate('click');
      expect(props.deleteReport.mock.calls.length).toBe(1);
    });

    it('should call "onClose" function on click of cancel and close sign', () => {
      wrapper.find('.button--dark').simulate('click');
      wrapper.find('.icon-close').simulate('click');
      expect(props.onClose.mock.calls.length).toBe(2);
    });
  });
});
