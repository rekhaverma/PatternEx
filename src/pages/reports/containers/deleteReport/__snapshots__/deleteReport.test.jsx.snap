// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteReport /> modal:  render:  should match with snapshot 1`] = `
<Modal
  className="ptrx-modal"
  style={Object {}}
  type=""
>
  <section
    className="deleteReport"
  >
    <div
      className="deleteReport__row +spaceBetween firstrow"
    >
      <span
        className="deleteReport__title"
      >
        <FormattedMessage
          id="reports.deleteReport.title"
          values={
            Object {
              "reportName": undefined,
            }
          }
        />
      </span>
      <span
        className="icon-close"
        onClick={[Function]}
      />
    </div>
    <div
      className="deleteReport__row +spaceBetween"
    >
      <FormattedMessage
        id="reports.deleteReport.confirmation"
        values={
          Object {
            "reportName": undefined,
          }
        }
      />
    </div>
    <div
      className="deleteReport__row +rightSide"
    >
      <Button
        className="button--success +small"
        onClick={[Function]}
      >
        <FormattedMessage
          id="reports.delete"
          values={Object {}}
        />
      </Button>
      <Button
        className="button--dark +small"
        onClick={[Function]}
      >
        <FormattedMessage
          id="reports.cancel"
          values={Object {}}
        />
      </Button>
    </div>
  </section>
</Modal>
`;
